rep-0151.rst

```diff
@@ -0,0 +1,293 @@
+
+Users building ROS from source will not have an existing ROS workspace set, so
+the variable won't be set, and conditional dependencies can't be evaluated
+when using ``rosdep`` to install them.
+To fix this, ``rosdep`` will set ``ROS_PYTHON_VERSION``.
+It must choose a Python version to match the ROS distro while allowing platforms
+that already use Python 3 to override it
+
+The choice is made using the following information:
+
+ * the command line option ``--rosdistro``
+ * the environment vairable ``ROS_DISTRO``
+ * the environment vairable ``ROS_PYTHON_VERSION``
+ * the value ``python_version`` in the ROS distribution file [8]_
+ * the version of Python used to invoke ``rosdep`` via ``sys.version[0]``
+
+If ``--rosdistro`` is set then the value of ``python_version`` in the ROS distro
+file is used.
+For the purpose of evaluating conditional dependencies, ``ROS_PYTHON_VERSION``
+will be set to ``python_version``, and ``ROS_DISTRO`` will be set to the value
+passed in via ``--rosdistro``.
+This allows users who may be unaware they have an existing ROS workspace sourced
+to install dependencies for a different ROS distro.
+
+Otherwise, if ``ROS_PYTHON_VERSION`` is set then that value is used.
+The value of ``ROS_DISTRO`` is unchanged whether it is set or unset.
+This allows users to install dependencies for a ROS distro with a different
+Python version than the ROS distro targets, such as Arch Linux users using
+Python 3 for ROS Melodic.
+
+If ``ROS_PYTHON_VERSION`` is not set but ``ROS_DISTRO`` is then
+for the purpose of evaluating conditional dependencies ``ROS_PYTHON_VERSION``
+will be set to the value of ``python_version``.
+This makes sure ``ROS_PYTHON_VERSION`` dependencies get evaluated in a way that
+matches the desired distro.
+This case should not happen so long as the user has sourced a workspace with the
+``ros_environment`` package, since that package sets both ``ROS_DISTRO`` and
+``ROS_PYTHON_VERSION``.
+
+Finally, if none of ``--rosdistro``, ``ROS_DISTRO`` or ``ROS_PYTHON_VERSION`` are set
+then for the purpose of evaluating conditional dependencies ``ROS_PYTHON_VERSION``
+will be set to ``sys.version[0]``.
+This makes sure ``ROS_PYTHON_VERSION`` dependencies are not silently ignored,
+though there is a risk of installing the wrong packages.
+
+rosdep and pip packages
+'''''''''''''''''''''''
+Unlike Debian packages, keys that resolve to packages on PyPI may refer to
+either Python 2 or Python 3 dependencies.
+The version that gets downloaded depends on the version of Python used to
+invoke pip.
+Rosdep currently calls the command ``pip``.
+It might use Python 2 when using the Debian package ``python-pip``, but it could
+use Python 3 inside a venv.
+``rosdep`` should pick a version of pip that matches ``ROS_PYTHON_VERSION``
+regardless of wheter a Python virtual env is being used.
+
+First ``rosdep`` should try the commands ``pip2`` or ``pip3`` since those are
```

Comment:
Has this been implemented? If yes, rephrasing the sentence would be good as well as adding a reference.

Reply:
Yes, ros-infrastructure/rosdep#709 , changed tense in 55a6e23bd43d88219c14a6bb8a009d50c54d879d